### PR TITLE
fix(ui): Ensure projects are loaded before setting GlobalSelectionStore

### PR DESCRIPTION
**Pull request by davidenwang**
https://sentry.io/organizations/sentry/issues/1309332036/?project=11276&query=is%3Aunassigned+is%3Aunresolved+level%3Aerror+%21error.type%3ABadGatewayError++%21error.type%3AForbiddenError+%21error.type%3AUnauthorizedError++%21error.type%3ACancelledError+%21error.type%3ANotFoundError+%21error.type%3ABadRequestError+%21error.type%3AGatewayTimeoutError+%21error.type%3ATooManyRequestsError

This error will happen when a user starts in the lighweight org context (projects page) which will trigger a request for /teams. If the user then navigates to a heavyweight org while the /projects haven't loaded, a new request for a detailed org will be requested and the sentry loading gif will show. During this time if the /teams endpoint were to return then the app will wrongly assume that the detailed org context has been returned when the org context that has come back only has teams and no projects.

TLDR: Check for the existence of projects before using it in global selection store.

### CODE REVIEW

**Comment by dashed** [comment]
I'm not familiar with the terminology of "lightweight org contexts" and "heavyweight org contexts". 

My best guess is that `teams` and `projects` gets polluted into `organization`, based on what I see here https://github.com/getsentry/sentry/blob/6eaecc3310bacf72fba717e2e0b74017fe1c6890/src/sentry/static/sentry/app/stores/organizationStore.jsx#L70-L86

Would it be useful to have a utility function to check if `organization` has `teams` or `projects`?

**Review by dashed — APPROVED** [review]
Looks good to me 👍 

The breadcrumbs make it the obvious.

**Review by billyvg — APPROVED** [review]

**Comment by davidenwang** [comment]
> I'm not familiar with the terminology of "lightweight org contexts" and "heavyweight org contexts".
> 
> My best guess is that `teams` and `projects` gets polluted into `organization`, based on what I see here
> 
> https://github.com/getsentry/sentry/blob/6eaecc3310bacf72fba717e2e0b74017fe1c6890/src/sentry/static/sentry/app/stores/organizationStore.jsx#L70-L86
> 
> Would it be useful to have a utility function to check if `organization` has `teams` or `projects`?
Yup you are correct, a heavyweight organization is just an organization hydrated with all projects and all teams. This is currently what most of the application assumes the `organization` object to be like. However most of the work I'm doing is making the webapp compatible with a lightweight organization which doesn't have projects or teams loaded in so that the app is not blocked waiting for all projects and all teams to be retrieved even when the information is not immediately shown.

It could be useful to have a function to check for the existence of teams or projects and I already have something similar here: https://github.com/getsentry/sentry/blob/f5e1cfeba1840e54a74b2a5cc6a5d98504783894/src/sentry/static/sentry/app/views/organizationContext.jsx#L131

but for this it's probably fine as the function would probably just be like `return !!(organization.projects) && !!(organization.teams)` so I don't think it's all that necessary.
